src/main/java/edu/itu/cavabunga/core/entity/component/Alarm.java

```diff
@@ -1,13 +1,27 @@
 package edu.itu.cavabunga.core.entity.component;
 
 import edu.itu.cavabunga.core.entity.Component;
+import edu.itu.cavabunga.core.entity.Property;
+import edu.itu.cavabunga.exception.Validation;
 
 import javax.persistence.Entity;
 
 @Entity
 public class Alarm extends Component{
     @Override
-    public boolean validate(){
-        return true;
+    public void validate(){
```

Comment:
new methods have implemented on Component super class for validation to prevent separated validation for every child class
validateOptionalOneProperties
validateRequiredOneProperties